pkg/operator/terminationobserver/termination_metrics.go

```diff
@@ -0,0 +1,57 @@
+package terminationobserver
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+type terminationMetrics struct {
+	store           Store
+	apiServerMetric *prometheus.GaugeVec
+}
+
+// RegisterMetrics registers the termination metrics into legacy prometheus registry.
+func RegisterMetrics(store Store) {
+	legacyregistry.MustRegister(newTerminationMetrics(store))
+}
+
+func newTerminationMetrics(store Store) *terminationMetrics {
+	return &terminationMetrics{
+		store: store,
+		apiServerMetric: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "openshift_kube_apiserver_termination",
+			Help: "Report individual termination events and their timestamps for each Kubernetes API server",
+		}, []string{"name", "createdTimestamp", "eventName"}),
+	}
+}
+
+var _ metrics.Registerable = &terminationMetrics{}
+
+func (t *terminationMetrics) Describe(ch chan<- *prometheus.Desc) {
+	ch <- t.apiServerMetric.WithLabelValues("").Desc()
+}
+
+func (t *terminationMetrics) Collect(ch chan<- prometheus.Metric) {
```

Comment:
metrics can be set imperatively. I don't see a reason to build this callback thing here which read an imperatively constructed state. Make it direct and let the prometheus library do the serving/collection.

Reply:
We use this pattern everywhere and I think this is the right way to expose metrics to prometheus. Having this baked into a controller and then determining when to send the data to prometheus seems like less error prone to me.